sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorSession.java

```diff
@@ -743,24 +808,71 @@ private <T extends AmqpLink> boolean removeLink(ConcurrentMap<String, LinkSubscr
         }
     }
 
+    private AmqpErrorContext getErrorContext() {
+        return protonSession.getErrorContext();
+    }
+
+    /**
+     * Obtain a mono that await for the session to be active with a timeout.
+     * <p>
+     * When a subscription is made to the mono, it verifies the most recent state reported by the {@code endpointStates}
+     * flux. The state can transition from none to active to error or completed Or it can also change from none to error
+     * or completed directly. If the state is not active yet, then the subscription waits for {@code tryTimeout} before
+     * timing out with a retriable {@link AmqpException}. If the state is error, then the error is returned, if the state
+     * transition to completed then a {@link AmqpException} that can be retried is returned.
+     * </p>
+     *
+     * @param protonSession the underlying {@link ProtonSession}.
+     * @param tryTimeout the duration to wait for the session to be active.
+     * @param endpointStates the flux streaming session endpoint states.
+     * @return a mono that completes when the session is active.
+     */
+    private static Mono<Void> activeAwaiter(ProtonSessionWrapper protonSession, Duration tryTimeout,
+        Flux<AmqpEndpointState> endpointStates) {
+        final String connectionId = protonSession.getConnectionId();
+        final String sessionName = protonSession.getName();
+        // session-active-timeout = session-open-timeout + 2 seconds buffer. (anu - double check buffer need?)
+        final Duration activeTimeout = tryTimeout.plusSeconds(2);
```

Comment:
2 seconds? I was wondering why that specific value?

Reply:
updated to use `tryTimeout`.